app/src/main/java/de/westnordost/streetcomplete/quests/AbstractQuestAnswerFragment.java

```diff
@@ -255,10 +255,12 @@ private Resources getEnglishResources()
 	{
 		if (!hasChanges())
 		{
+			onDiscard();
 			confirmed.run();
 		}
 		else
 		{
+			onDiscard();
```

Comment:
💥 : onDiscard must be in the onYes callback, otherwise it is called **before** the user acknowledges discarding the message.